### PR TITLE
Move self_shl & self_shr to their respective callable version

### PR DESCRIPTION
**Pull request by SadiinsoSnowfall**
+ added tests for assigning shifts

### CODE REVIEW

File: include/eve/detail/assert_utils.hpp [hunk 1]
```diff
@@ -14,7 +14,7 @@
 namespace eve::detail
 {
   template<typename A0, eve::value A1>
-  constexpr EVE_FORCEINLINE bool assert_good_shift(A1 const &t) noexcept
+  constexpr EVE_FORCEINLINE bool assert_good_shift(A1 t) noexcept
```
Comment (DenisYaroshevskiy) [inline include/eve/detail/assert_utils.hpp:17]: I find it weird that we allow a shift by negative number for signed value type.

gcc complains: https://godbolt.org/z/hEdT5zfed

Not for this pr though.

**Review by DenisYaroshevskiy — COMMENTED** [review]
lgtm

File: include/eve/detail/assert_utils.hpp (hunk 1 above)
Comment (DenisYaroshevskiy) [reply include/eve/detail/assert_utils.hpp]: You either go all signed shifts or none signed shifts. The "signed only for signed numbers" makes no sense.

**Comment by DenisYaroshevskiy** [comment]
Iota test is correct. It shows a real problem in the system. Don't put asserts

**Comment by jfalcou** [comment]
The decision we took for now is to remove assert from shifts. We can advance on that, finish the callable transition and suppport for type translation and think about how EVE shoudl sue assert in the mean time.

@DenisYaroshevskiy are you OK with that ?

**Pull request merged**